dijet/tasks/alpha.py

```diff
@@ -0,0 +1,196 @@
+        process_names = [item.name for item in process_sets]
+        # Get all samples from input belonging to process
+        samples = set(self.datasets).intersection(process_names)
+        if len(samples) == 0:
+            samples = process_names
+        return (
+            samples,
+            self.config_inst.get_dataset(process_sets[0]).is_mc,
+        )
+
+    def output(self) -> dict[law.FileSystemTarget]:
+        # TODO: Unstable for changes like data_jetmet_X
+        #       Make independent like in config datasetname groups
+        sample = self.extract_sample()
+        target = self.target(f"{sample}", dir=True)
+        # declare the main target
+        outp = {
+            "alphas": target.child("alphas.pickle", type="f"),
+            "asym": target.child("asym.pickle", type="f", optional=True),
+        }
+        return outp
+
+    def get_norm_asymmetries(self, histogram, method):
+
+        # NOTE: Alternative loop over alpha like here:
+        #       histogram[hist.loc(method), slice(hist.loc(0), hist.loc(0.3), sum), :, :, :].values()
+        # TODO: Loose hist structure here. Methd to keep structure here ?
+        #       histogram[hist.loc(method), slice(hist.loc(0), hist.loc(0.3)), :, :, :]
+        #       For integral not taking .values()
+        values = histogram[hist.loc(method), slice(hist.loc(0), hist.loc(0.3)), :, :, :].values()
+        # axis = 0 alpha
+        inclusiv = np.apply_along_axis(np.cumsum, 0, values)
+        # axis = 3 asymmetry
+        integral = inclusiv.sum(axis=3, keepdims=True)
+
+        # Store in dictonary to use in alpha extrapolation
+        return {"content": inclusiv, "integral": integral}
+
+    def process_asymmetry(self, hists, asyms):
+
+        inclusive_norm = hists["content"] / hists["integral"]
+        means = np.nansum(asyms * inclusive_norm, axis=3, keepdims=True)
+
+        # TODO: np.nanaverage ?
+        #       -> only numpy.nanmean but no weights
+        stds = np.sqrt(np.average(((asyms - means)**2), weights=inclusive_norm, axis=3))
+        stds_err = stds / np.sqrt(np.squeeze(hists["integral"]))
+
+        return {"widths": stds, "errors": stds_err}
+
+    def method_index(self, method):
+        indices = {"sm": 1, "fe": 2}
+        # TODO: check that method is either sm or fe
+        return indices[method]
+
+    def run(self):
+        # TODO: Gen level for MC
+        #       Correlated fit (in jupyter)
+
+        h_all = []
+        datasets, isMC = self.get_datasets()
+
+        for dataset in datasets:
+            for variable in self.variables:
+                h_in = self.reduce_histogram(
+                    self.load_histogram(dataset, variable),
+                    self.processes,
+                    self.shift,
+                )
+                # Store all hists in a list to sum over after reading
+                h_all.append(h_in)
+        h_all = sum(h_all)
+        amax = np.array([0.05, 0.1, 0.15, 0.2, 0.25, 0.3])
```

Comment:
I would keep it for now. The edges go from 0-1, but we only want to go to 0.3 since here the linear dependency (normally) still holds. This topic will be addressed in a next PR when we move to a histogram based structure and define the limit (0.3) in the config.